modules/subscriber/main.tf

```diff
@@ -0,0 +1,10 @@
+locals {
+  has_discovery_rate          = var.discovery_rate != ""
```

Comment:
You can assume that discovery rate must be set for this module, so you don't need this local variable.

Reply:
i thought you can instantiate a subscriber without actually having a discovery rate? and the subscriber lambda just exists and you can post subscription requests to it